yoticlient.go

```diff
@@ -97,25 +92,28 @@ func (client *Client) GetSdkID() string {
 	return client.SdkID
 }
 
-// GetActivityDetails requests information about a Yoti user using the one time use token generated by the Yoti login process.
-// It returns the outcome of the request. If the request was successful it will include the users details, otherwise
-// it will specify a reason the request failed.
-func (client *Client) GetActivityDetails(token string) (ActivityDetails, []string) {
-	_, activity, errStrings := client.getActivityDetails(token)
-	return activity, errStrings
+// GetActivityDetails requests information about a Yoti user using the one time
+// use token generated by the Yoti login process. It returns the outcome of the
+// request. If the request was successful it will include the user's details,
+// otherwise an error will be returned, which will specify the reason the
+// request failed. If the function call can be reattempted with the same token
+// the error will be a temporary error.
+func (client *Client) GetActivityDetails(token string) (ActivityDetails, error) {
+	_, activity, err := client.getActivityDetails(token)
+	return activity, err
 }
 
-func (client *Client) getActivityDetails(token string) (userProfile UserProfile, activity ActivityDetails, errStrings []string) {
+func (client *Client) getActivityDetails(token string) (userProfile UserProfile, activity ActivityDetails, err error) {
 
 	httpMethod := http.MethodGet
 	key, err := loadRsaKey(client.Key)
 	if err != nil {
-		errStrings = append(errStrings, fmt.Sprintf("Invalid Key: %s", err.Error()))
+		err = fmt.Errorf("Invalid Key: %s", err.Error())
 		return
 	}
 	token, err = decryptToken(token, key)
 	if err != nil {
-		errStrings = append(errStrings, fmt.Sprintf("Invalid Key: %s", err.Error()))
+		err = fmt.Errorf("Invalid Key: %s", err.Error())
```

Comment:
Invalid Key looks like the wrong error message here.

It would also be good to have a test that checks this error case, and I guess it would assert that the error is not temporary, since this is a failure to decrypt a token which feels pretty permanent.